src/main/java/core/basesyntax/strategy/SupplyOperation.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.ShopTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handle(ShopTransaction shopTransaction) {
+        String fruit = shopTransaction.getFruitName();
+        int weight = Storage.fruitShopStorage.get(fruit);
```

Comment:
There is a potential `NullPointerException` here if the fruit does not exist in the storage. Consider using `Storage.fruitShopStorage.getOrDefault(fruit, 0)` to safely handle cases where the fruit is not yet in the storage.